### PR TITLE
adds CircleCI webhook integration

### PR DESCRIPTION
**Pull request by vitalis**
* adds CircleCI webhook integration
* Add any screenshots if you are touching the UI
* Remember to add unit tests

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link:

### Risks
- Level: Low/Med/High

### CODE REVIEW

**Comment by grosser** [comment]
why ?

**Comment by vitalis** [comment]
@grosser 
We are using CircleCI as our CI, It will be helpful addition for us and others that use Circle

File: test/controllers/integrations/circleci_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  end
+
+  before { Deploy.delete_all }
+
+  test_regular_commit "Circleci", failed: {payload: {status: "failed"}}, no_mapping: {payload: {branch: "foobar"}} do
+    project.webhooks.create!(stage: stages(:test_staging), branch: "master", source: 'circleci')
+  end
+
+  describe "skipping" do
+    it "doesn't trigger a deploy if we want to skip with [deploy skip]" do
```
Comment (grosser) [inline test/controllers/integrations/circleci_controller_test.rb]: this test is copied twice ? ... did you want to test something else ?

**Comment by grosser** [comment]
👍 looks good / clean :)

File: test/controllers/integrations/circleci_controller_test.rb (hunk 1 above)
Comment (vitalis) [inline test/controllers/integrations/circleci_controller_test.rb]: One is testing `[deploy skip]` the other is testing `[skip deploy]` took it from:
https://github.com/zendesk/samson/blob/master/test/controllers/integrations/semaphore_controller_test.rb

Comment (grosser) [inline test/controllers/integrations/circleci_controller_test.rb]: ah ... did not see that :D

**Comment by grosser** [comment]
cleanup https://github.com/zendesk/samson/pull/1179

**Pull request merged**